schema/workflow.yaml

```diff
@@ -325,357 +394,435 @@ $defs:
+                    title: ContainerPorts
                     description: The container's port mappings, if any.
                   volumes:
                     type: object
+                    title: ContainerVolumes
                     description: The container's volume mappings, if any.
                   environment:
-                    title: ContainerEnvironment
                     type: object
+                    title: ContainerEnvironment
                     description: A key/value mapping of the environment variables, if any, to use when running the configured process.
                 required: [ image ]
             required: [ container ]
-            description: Enables the execution of external processes encapsulated within a containerized environment.
           - title: RunScript
+            description: Enables the execution of custom scripts or code within a workflow, empowering workflows to perform specialized logic, data processing, or integration tasks by executing user-defined scripts written in various programming languages.
             properties:
               script:
                 type: object
+                title: Script
+                description: The configuration of the script to run.
                 unevaluatedProperties: false
                 properties:
                   language:
                     type: string
+                    title: ScriptLanguage
                     description: The language of the script to run.
                   arguments:
-                    title: ScriptArguments
                     type: object
-                    additionalProperties: true
+                    title: ScriptArguments
                     description: A key/value mapping of the arguments, if any, to use when running the configured script.
+                    additionalProperties: true
                   environment:
-                    title: ScriptEnvironment
                     type: object
-                    additionalProperties: true
+                    title: ScriptEnvironment
                     description: A key/value mapping of the environment variables, if any, to use when running the configured script process.
+                    additionalProperties: true
                 oneOf:
                   - title: ScriptInline
+                    type: object
+                    description: The script's code.
                     properties:
                       code:
                         type: string
                     required: [ code ]
-                    description: The script's code.
                   - title: ScriptExternal
+                    type: object
+                    description: The script's resource.
                     properties:
                       source:
                         $ref: '#/$defs/externalResource'
-                    description: The script's resource.
+                        title: ScriptExternalSource
```

Comment:
No necessarily external, even though I see what you meant with that, but it might be confusing to users when using a "local", relative source. Why not ScriptSource?

Reply:
`ScriptExternal` was already present in the document before this PR, by opposition to the `Inline` one I guess.

The "inline" one is also the "source" of the script, but refered to as `code` so using `ScriptSource` could be confusing as well.

I suggest renaming:
- `ScriptInline` with `InlineScript` (and add the missing title `InlineScriptCode` for the `code` property)
- `ScriptExternal` with `ExternalScript`
- therefore `ScriptExternalSource` with `ExternalScriptResource`

WDYT ?